### DISCUSSION TITLE
Network bandwidth usage?

### DISCUSSION THREAD

**Discussion by djch** [django-q2/django-q2#141] [Q&A] [answered]
Hi all,

Our PaaS provider recently started charging for network egress fees. I have noticed that Django Q2 seems to use quite a lot of network bandwidth even when "idle". In this instance, Q is running some very low-intensity daily tasks which send email reminders via a web API.

Here's a graph of network activity over one week (each datapoint represents a one-hour period):
![CleanShot 2023-11-22 at 11 08 20@2x](https://github.com/django-q2/django-q2/assets/4392286/7cc4aa6e-7025-4362-b7de-bf316180bdf9)

That's accumulated to something like ~36GB so far during November.

Our config looks like this:
```python
Q_CLUSTER = {
    'name': 'project_name',
    'label': 'Tasks',
    'catch_up': False,
    'orm': 'default',
    'timeout': 2700,
    'retry': 3600,
}
```

Is this normal for something like Django Q?

Given our very light usage, would we benefit from setting a much higher `poll` rate or something?

**Comment by GDay** [comment]
Do you think that's all from django-q2? Because that sounds very high.

Django-q2 does poll the database quite often for new tasks. You can indeed change the `poll` rate, that should reduce the amount of bandwidth being used. Default is 0.2. You could easily up it to 30 in your case.

**Reply by djch** [reply]
Thanks. I've upped the `poll` to 60 and it seems to be making some difference so far:
<img width="955" alt="Screenshot 2023-11-22 at 12 14 14 pm" src="https://github.com/django-q2/django-q2/assets/4392286/151274e7-09b7-49d2-8077-cc1a21800a30">

This particular process just runs `q_cluster` so all the bandwidth represented in that graph would be from that, yeah.

I've just pushed another change which will reduce the `workers` to 1, instead of using the default number of vCPUs. Hopefully that might improve this slightly too (and the memory usage).

**Reply by GDay** [reply]
You could also try to set the [guard_cycle](https://django-q2.readthedocs.io/en/master/configure.html#guard-cycle) if you want to reduce memory use/cpu use. It should also help with bandwidth usage as it saves the current state of the cluster every time it runs through a cycle (which is currently every 0.5 seconds). Default is 0.5 seconds and should be between 0 and 60 seconds.

**Reply by djch** [reply]
Increasing the `guard_cycle` to 20s had a massive impact on the network utilisation. Thanks for that tip. 

And our two daily scheduled tasks ran without issue this morning.

<img width="979" alt="Screenshot 2023-11-23 at 9 22 20 am" src="https://github.com/django-q2/django-q2/assets/4392286/9758e15a-d1b8-4665-b0f5-59fb7c6a5308">

**Reply by GDay** [reply, marked as answer]
Thanks a lot for reporting back. I wasn't aware that django-q(2) is so heavy on bandwidth calls. 36GB is absolutely insane. 

The status saving on the guard cycle is in most cases excessive, especially with very few schedules/tasks. I will rewrite that to first check (compared to a local object) if we actually need to update the status. With that, bandwidth would likely get even lower for you as it would only need to update 2 times a day for your daily scheduled tasks, instead of every 20 seconds now.
